persist/persist_test.go

```diff
@@ -0,0 +1,25 @@
+package persist
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/build"
+)
+
+// TestIntegrationRandomSuffix checks that the random suffix creator creates
+// valid files.
+func TestIntegrationRandomSuffix(t *testing.T) {
```

Comment:
Yeah that definition is a little odd. To me, a pure (i.e. stateless) function is exactly the sort that you'd want to run unit tests on. My view is:
- Unit test: single function being tested.
- Integration test: multiple functions interacting with each other.

Both types of test may requiring poking around in the internal state. The important thing is that unit tests are testing a single "unit" of code, either an entire function or a well-defined and easily separable piece of code.

Obviously, unit tests are easier than integration tests, which is why it's advantageous to use as many pure functions as possible.